indirect_with_allocators.h

```diff
@@ -0,0 +1,106 @@
+#ifndef XYZ_INDIRECT_WITH_ALLOCATORS_H
+#define XYZ_INDIRECT_WITH_ALLOCATORS_H
+#include <memory>
+#include <utility>
+
+namespace xyz {
+template <class T, class A = std::allocator<T>>
+class indirect {
+  T* p_;
+  [[no_unique_address]] A alloc_;
```

Comment:
MSVC ignores `[[no_unique_address]]` but they have their own ``[[msvc::no_unique_address]]` which will be necessary here for MSVC to pass a test that `sizeof(indirect<T>) == sizeof(T*)`